packages/project/src/db/project.ts

```diff
@@ -246,3 +246,21 @@ export const updateDomain = async (
     throw error;
   }
 };
+
+export const setMarketplaceApprovalStatus = async (
+  {
+    projectId,
+    marketplaceApprovalStatus,
+  }: {
+    projectId: Project["id"];
+    marketplaceApprovalStatus: MarketplaceApprovalStatus;
+  },
+  context: AppContext
+) => {
+  await assertEditPermission(projectId, context);
```

Comment:
Looks wrong, as in this case anyone with edit permission can set any `marketplaceApprovalStatus`
but the idea that final status can be set only by predefined lists of moderators.

Reply:
done